centreon/www/include/configuration/configObject/contact/DB-Func.php

```diff
@@ -197,6 +197,42 @@ function disableContactInDB($contact_id = null, $contact_arr = array())
     }
 }
 
+/**
+ * Unblock contacts in the database
+ *
+ * @param int|array|null $contact Contact ID, array of contact IDs or null to unblock all contacts
+ */
+function unblockContactInDB($contact = null)
+{
+    global $pearDB, $centreon;
+
+    if (is_null($contact)) {
+        return;
+    }
+
+    if (!is_array($contact)) {
+        $contact = [$contact => "1"];
+    }
+
+
+    $contactIds = array_map('intval', array_keys($contact));
+    $idPlaceholders = implode(',', array_fill(0, count($contactIds), '?'));
+
+
+    //updating the given users
+    $updateQuery = "UPDATE contact SET blocking_time = null WHERE contact_id IN ($idPlaceholders)";
+    $statement = $pearDB->prepare($updateQuery);
+    $statement->execute($contactIds);
+
+    // retrieve the users and add log
+    $selectQuery = "SELECT contact_id, contact_name FROM contact WHERE contact_id IN ($idPlaceholders)";
+    $statement = $pearDB->prepare($selectQuery);
+    $statement->execute($contactIds);
```

Comment:
To avoid binding with question mark you can use the following pattern:

```suggestion
$bindContactIds = [];
foreach (array_keys($contact) as $contactId) {
    $bindContactIds[':contact_' . $contactId] = $contactId;
}

// ':contact_1, :contact_2, :contact_3' etc ...
$idPlaceholders = implode(', ', array_keys($bindContactIds));

// retrieve the users and add log
$updateQuery = "UPDATE contact SET blocking_time = null WHERE contact_id IN ($idPlaceholders)";
$statement = $pearDB->prepare($updateQuery);
foreach ($bindContactIds as $token => $value) {
    $statement->bindValue($token, $value, \PDO::PARAM_INT);
}
$statement->execute();

// retrieve the users and add log
$selectQuery = "SELECT contact_id, contact_name FROM contact WHERE contact_id IN ($idPlaceholders)";
$statement = $pearDB->prepare($selectQuery);
foreach ($bindContactIds as $token => $value) {
    $statement->bindValue($token, $value, \PDO::PARAM_INT);
}
$statement->execute();

```

first you create an array with all your :contact_ token (:contact_<contact_id>) and the id as a value, then you can implode  your keys to create a string of all those tokens.
Finally you can iterate on your array to call your bindValue

iteration 1:
:contact_1, 1, \PDO::PARAM_INT

iteration 2:
:contact_2, 2, \PDO::PARAM_INT

etc